### ISSUE TITLE
Update wiki page with another method for making SVGs WCAG compliant

### ISSUE THREAD

**Issue by roslynwythe** [hackforla/website#6653]
### Overview
As developers we need to update a wiki page with an additional method for providing SVG graphics with accessible names in order to comply with WCAG

### Details 
An accessible name is a short string of text that identifies a user interface element and gives users of assistive technology a label for it.  The "alt-text" attribute is used to provide an accessible name for an img element, however the "alt-text" attribute is not applicable to an svg element.  The wiki page https://github.com/hackforla/website/wiki/How-to-make-SVGs-and-other-images-WCAG-compliant describes methods for providing an accessible name to svg elements, and in this issue we will add a new method.  

### Action Items
- [x] We are currently in the process of moving the old wiki to the new [website-wiki repo](https://github.com/hackforla/website-wiki), so we will not be making any changes or additions to the old wiki at this time. Thus, we will be adding wiki content through a different process now. Read [How to Contribute to the Wiki](https://github.com/hackforla/website/wiki/How-to-Contribute-to-the-Wiki)
- [x] Following the process outlined above in [How to Contribute to the Wiki](https://github.com/hackforla/website/wiki/How-to-Contribute-to-the-Wiki), copy the raw source of the page https://github.com/hackforla/website/wiki/How-to-make-SVGs-and-other-images-WCAG-compliant into a comment in this issue
- [x] Add a new section immediately before the text ""Accessible names/descriptions" with the heading "Use an outer svg with aria-label or aria-labelledby"
- [x] Mention that in many instances, we wrap svg elements in outer svg elements, in order to apply styling attributes (such as `fill` and `viewBox` to the outer svg), and outer svg can also provide an accessible name using these rules:
   - [x] If the desired accessible name appears as text in another element on the page, use the `aria-labelledby` attribute to reference the element containing the desired text.  
 ```
Example:
<svg role="img"  viewBox="0 0 100 100"  aria-labelledby="resource-title">
   {% include svg/icon-github-color.svg %}
<svg>
<h4 id="resource-title">GitHub</h4>
```
   - [x] If there is no element displaying the desired accessible name, use the `aria-label` attribute to specify the name, or else use the `aria-labelledby` attribute and reference a new child `<title>` element.  If used, the `<title>` element should be the first child of the outer `<svg>`  
```
Example:
<svg role="img"  viewBox="0 0 100 100"  aria-labelledby="titleID">
   <title id="titleID">GitHub</title>
   {% include svg/icon-github-color.svg %}
<svg>
```
- [x] Mention that if an appropriate name exists on the page, use of `aria-labelledby` is preferable to `aria-label`

### Resources/Instructions

<details><summary>HTML and accessibility tree depicting a duplicate accessible name "GitHub GitHub" on the link wrapping an svg element</summary>

![elements outer svg but wrapping div has no role - dup](https://github.com/hackforla/website/assets/5314153/4909848c-aeb4-43cd-80eb-9b7c079aa1ff)

![tree - outer svg but wrapping div has no role - result is dup](https://github.com/hackforla/website/assets/5314153/c77aa5ea-212a-4735-8d7d-7da1dfbb5207)

</details> 

<details><summary>HTML and accessibility tree depicting solution to duplicate accessible name</summary>

![elements using aria-labelledby on link to remove dup](https://github.com/hackforla/website/assets/5314153/05c1cc73-7703-459b-8492-27e89ddf911a)

![using aria labelledby in link to fix dup name](https://github.com/hackforla/website/assets/5314153/4860cf8f-0b69-4152-8cc9-49643e8284e1)

</details> 

https://github.com/hackforla/website/wiki/How-to-make-SVGs-and-other-images-WCAG-compliant
https://www.w3.org/TR/using-aria/
https://canvas.workday.com/accessibility/the-accessibility-tree/#how-to-view-the-accessibility-tree
- This issue resulted from  #4242

**Comment by Jperparas** [comment]
**Availability:** 4/25-4/30 early afternoon

**ETA:** 4/30 EoD

**Comment by Jperparas** [comment]
Overview
Using the `alt` property in `img` HTML tags or Accessible Rich Internet Applications ([ARIA](https://www.w3.org/WAI/ARIA/apg/practices/names-and-descriptions/)) labels with Scalable Vector Graphics (SVGs) and other images is important so that all images on the Hack for LA website are Web Content Accessibility Guidelines ([WCAG](https://www.w3.org/WAI/standards-guidelines/wcag/)) compliant. This pages explains how the Website Team deals with SVGs and other images to make them WCAG compliant.

## Methods

### Use an `img` HTML tag with the SVG or image as for the `src` and an `alt` attribute
#### When to use it
- Any new images added to the website repo, which are added to [/assets/images](https://github.com/hackforla/website/tree/gh-pages/assets/images) directory or in a subdirectory of it
- Images already in [/assets/images](https://github.com/hackforla/website/tree/gh-pages/assets/images) directory or in a subdirectory of it

#### Notes
- For SVGs, see the pros and cons detailed in [_The quick way: img element_](https://developer.mozilla.org/en-US/docs/Learn/HTML/Multimedia_and_embedding/Adding_vector_graphics_to_the_Web#the_quick_way_img_element) section in MDN Web Docs _Adding vector graphics to the web_.

#### Example of an image in an `img` tag that needs an `alt` attribute: 
```
<img class="header-hero-image" src="/assets/images/toolkit/toolkit-hero.svg">
```
#### Steps:
In an `img` HTML tag, 
1. The image needs to be in the [/assets/images](https://github.com/hackforla/website/tree/gh-pages/assets/images) directory or be in a subdirectory of it  
2. Set `src` to the path to the SVG or image
3. Add an `alt` attribute with an appropriate descriptive text. (Depending on the image, the `alt` attribute will either be an empty `alt` attribute (`alt=""`) or will have an appropriate descriptive text. You can use this [alt Decision Tree](https://www.w3.org/WAI/tutorials/images/decision-tree/) to help you decide what the `alt` attribute should contain.)

```
<img class="header-hero-image" src="/assets/images/toolkit/toolkit-hero.svg" alt="" >
```

### Use an ARIA label in the `div` HTML tag wrapping the SVG
#### When to use it
- Any existing images in the [_includes](https://github.com/hackforla/website/tree/gh-pages/_includes) in the website repo

#### Notes
- For SVGs, see the pros and cons detailed in [_How to include SVG code inside your HTML_](https://developer.mozilla.org/en-US/docs/Learn/HTML/Multimedia_and_embedding/Adding_vector_graphics_to_the_Web#how_to_include_svg_code_inside_your_html) section in MDN Web Docs _Adding vector graphics to the web_. This section discusses inline SVG, which is what this method does using [Jekyll's include tag](https://jekyllrb.com/docs/includes/) to insert an SVG into a HTML file and the result of this is an inline SVG when the site is generated.

#### Example of an SVG wrapped in a `div` tag that needs an ARIA label: 
```
<div>
  {% include svg/logo-hfla-small.svg %}
</div>
```
#### Steps:
In an `div` HTML tag, 
1. Add a `role="img"`
2. Add an `aria-label` with an appropriate descriptive text.

```
<div role="img" aria-label="Hack for LA">
  {% include svg/logo-hfla-small.svg %}
</div>
```
### Use an outer SVG with `aria-label` or `aria-labelledby`

In many instances, we wrap svg elements in outer svg elements, in order to apply styling attributes (such as `fill` and `viewBox` to the outer svg), and outer svg can also provide an accessible name using these rules:

 - If the desired accessible name appears as text in another element on the page, use the `aria-labelledby` attribute to reference the element containing the desired text
 ```Example:
<svg role="img"  viewBox="0 0 100 100"  aria-labelledby="resource-title">
  {% include svg/icon-github-color.svg %}
</svg>
<h4 id="resource-title">GitHub</h4>
```
- If there is no element displaying the desired accessible name, use the `aria-label` attribute to specify the name, or else use the `aria-labelledby` attribute and reference a new child `<title>` element. If used, the `<title>` element should be the first child of the outer `<svg>`
> [!NOTE]
> If an appropriate name exists on the page, use of `aria-labelledby` is preferable to `aria-label`
```Example:
<svg role="img"  viewBox="0 0 100 100"  aria-labelledby="titleID">
   <title id="titleID">GitHub</title>
   {% include svg/icon-github-color.svg %}
</svg>
```


#### Accessible names/descriptions
Review the w3.org document about accessible names and descriptions [here](https://www.w3.org/WAI/ARIA/apg/practices/names-and-descriptions/).

> Naming:
> * Naming with child content.
> * Naming with a string attribute via aria-label.
> * Naming by referencing content with aria-labelledby.
> * Naming form controls with the label element.
> * Naming fieldsets with the legend element.
> * Naming tables and figures with captions.
> * Fallback names derived from titles and placeholders.

> Describing:
> * Describing by referencing content with aria-describedby.
> * Describing tables and figures with captions.
> * Descriptions derived from titles.


### Resources
* [MDN: aria-label](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-label)
* [MDN: ARIA, img role](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/img_role)
* [a11y-101: Accessible SVGs](https://a11y-101.com/development/svg)
* [w3.org: Accessible Names and Descriptions](https://www.w3.org/WAI/ARIA/apg/practices/names-and-descriptions/)
* [WCAG 2 Overview](https://www.w3.org/WAI/standards-guidelines/wcag/)

**Comment by roslynwythe** [comment]
@Jperparas this is well done, however the "Solution" under "Potential conflicts with adding an accessible name to an element" is not always effective and the situation is rather complex, so I think it would be better to omit mention of this problem.  I modified the last Action Item to be much simpler, so please edit accordingly.

**Comment by Jperparas** [comment]
@roslynwythe Thank you! I was dubious of the solution as well.  I omitted the potential conflict section and added the new last Action Item.

**Comment by Jperparas** [comment]
1. **Progress:**  Completed all Tasks.  linked comment to `Content to be added/edited in the new website-wiki repo`
2. **Blockers:** No Blockers waiting review
3. **Availability** Early Afternoon 5/11-5/13

**Comment by roslynwythe** [comment]
Thank you @Jperparas for creating this documentation of methods to make accesible SVGs.
